HDFOutputer.cc

```diff
@@ -0,0 +1,163 @@
+#include "HDFOutputer.h"
+#include "summarize_serializers.h"
+#include "lz4.h"
+#include <iostream>
+#include <cstring>
+#include <cmath>
+#include <set>
+
+constexpr int max_batch_size = 2; 
+
+using namespace cce::tf;
+using product_t = std::vector<char>; 
+
+HDFOutputer::HDFOutputer(std::string const& iFileName, unsigned int iNLanes) : 
+  file_(hdf5::File::create(iFileName.c_str())),
+  serializers_{std::size_t(iNLanes)},
+  serialTime_{std::chrono::microseconds::zero()},
+  parallelTime_{0}
+  {}
+
+HDFOutputer::~HDFOutputer() { }
+
+void HDFOutputer::setupForLane(unsigned int iLaneIndex, std::vector<DataProductRetriever> const& iDPs) {
+  auto& s = serializers_[iLaneIndex];
+  s.reserve(iDPs.size());
+  dataProductIndices_.reserve(iDPs.size());
+  for(auto const& dp: iDPs) {
+    s.emplace_back(dp.name(), dp.classType());
+  }
+   products_.reserve(10000);
```

Comment:
I'm pretty sure this can be
```suggestion
   products_.reserve(iPDs.size());
```